### PR TITLE
Soft transitions

### PR DESCRIPTION
**Pull request by hpmax**
I thought I had already committed part of this, that included a number of efficiency improvements, but apparently, I either didn't or no one accepted the changes.

The main thing in this new release is soft volume transitions which should remove clicks and pops.

Things I'd still like to consider doing but haven't done are modified waveforms and and fixed point.  Also as previously pointed out, the pulseaudio system is screwed up as run from services.

### CODE REVIEW

File: audiovario.c [hunk 1]
```diff
@@ -39,105 +39,141 @@ void vario_unmute(){
...
+
 	// init config struct
 	vario_config[vario].deadband_low = DEADBAND_LOW;
 	vario_config[vario].deadband_high = DEADBAND_HIGH;
-	vario_config[vario].pulse_length = PULSE_LENGTH;
-	vario_config[vario].pulse_length_gain = PULSE_LENGTH_GAIN;
-	vario_config[vario].pulse_duty = PULSE_DUTY;
+	// vario_config[vario].pulse_length = PULSE_LENGTH; // don't need this any more (see loval)
+	// vario_config[vario].pulse_length_gain = PULSE_LENGTH_GAIN; // don't need this any more (see hival)
+	// vario_config[vario].pulse_duty = PULSE_DUTY; // don't need this any more (see riseduty + risedutyfall)
```
Comment (kedder) [inline audiovario.c]: Please remove these lines if they are not needed

**Review by kedder — APPROVED** [review]
Looks good otherwise

**Comment by hpmax** [comment]
Okay, cleaned up the comments a bit.

Many of the parameters that we input to control the behavior are no longer actually used by the program, instead I'm now using them to pre-compute derivatives of the information to speed up processing in the audio generation loop.  So pulse_length, and pulse_length_gain are not directly used, but they are displayed upon initial run.  With those lines commented out, IF the values aren't specified in the .conf file, they would be displayed incorrectly, even though it wouldn't impact the actual program behavior.  So I put them back in.  Pulse_length_duty on the other hand is never used except to calculate other values so I deleted that line.

**Review by kedder — APPROVED** [review]
LGTM

**Pull request merged**